src/api/user.ts

```diff
@@ -0,0 +1,90 @@
+// Import Third-party Dependencies
+import * as httpie from "@myunisoft/httpie";
+
+// Import Internal Dependencies
+import * as utils from "../utils.js";
+
+interface NpmPaginated<T> {
+  total: number;
+  objects: T[];
+  urls: {
+    next: string;
+    prev: string;
+  }
+}
+
+interface NpmAvatars {
+  small: string;
+  medium: string;
+  large: string;
+}
+
+export interface NpmPackage {
+  id: number;
+  name: string;
+  description: string;
+  maintainers: string[];
+  version: string;
+}
+
+export interface PaginationParams {
+  perPage: number;
+  page: number;
+}
+
+export interface NpmUserProfile {
+  id: number;
+  name: string;
+  fullname?: string;
+  accounts: {
+    twitter?: string;
+    github?: string;
+  };
+  avatars: NpmAvatars
+  packages: NpmPaginated<NpmPackage>;
+  pagination: PaginationParams;
+}
+
+interface NpmWebUser {
+  scope: {
+    id: number;
+    parent: {
+      name: string;
+      avatars: NpmAvatars
+      resource: {
+        github?: string;
+        twitter?: string;
+        fullname?: string;
+      }
+    };
+  }
+  packages: NpmPaginated<NpmPackage>;
+  pagination: PaginationParams;
+}
+
+export async function user(
+  username: string,
+  { perPage, page }: Partial<PaginationParams> = { perPage: 25, page: 0 }
+): Promise<NpmUserProfile> {
```

Comment:
```suggestion
  params: Partial<PaginationParams> = {}
): Promise<NpmUserProfile> {
const { perPage = 25, page = 0 } = params;
```

I prefer this way.
If you do `await user("foo", { page = 0 })`, with your version `perPage` will be `undefined`, not this way

Reply:
yes, this was my intention as well, that defaults should apply even if only one of the two properties were set. My method seem to work as well, confirmed with this adhoc test

<img width="559" alt="Screenshot 2023-12-11 at 10 32 14 PM" src="https://github.com/NodeSecure/npm-registry-sdk/assets/49707819/e2bbd6bf-8f01-4241-a1b5-77e4da77ca3d">

Reply:
nope strike that, you're right. it was indeed using undefined in query if i log the params received and the url constructed. Will make the suggested change
<img width="631" alt="image" src="https://github.com/NodeSecure/npm-registry-sdk/assets/49707819/aa300825-b44f-4742-af2c-98d54b1812ed">

Reply:
Isn't it because of default value ?
![image](https://github.com/NodeSecure/npm-registry-sdk/assets/39910767/7b3206e5-c9ae-4487-acdd-a7565b199353)
What happend if you set per_page to 15 in the code ?